### PR TITLE
fix(aws): add missing base path to data routes

### PR DESCRIPTION
**Pull request by alebelcor**
Related https://github.com/opennextjs/opennextjs-aws/issues/670

I was trying to figure out how to add (and run) a unit test in `matcher.test.ts` but then I saw [this issue](https://github.com/opennextjs/opennextjs-aws/issues/539) that mentions all unit tests are broken (?).

If e2e test(s) are required I'd appreciate any help as I'm not very familiar. And let me know if you need something else.

Thanks!

### CODE REVIEW

**Comment by sommeeeer** [comment]
Unit tests should work again. I guess you could have one inside `describe("fixDataPage"....`.

**Review by conico974 — CHANGES_REQUESTED** [review]
No unit test run just fine.
In `packages/tests-unit` just run `pnpm run dev`.
You'll have to add a test here https://github.com/opennextjs/opennextjs-aws/blob/3f85a9027349b55c82995f28078b019211bc3205/packages/tests-unit/tests/core/routing/matcher.test.ts#L471 and modify NextConfig as in here https://github.com/opennextjs/opennextjs-aws/blob/3f85a9027349b55c82995f28078b019211bc3205/packages/tests-unit/tests/core/routing/matcher.test.ts#L228

**Comment by alebelcor** [comment]
Added two tests. Test suite seems to run successfully.

Let me know if those are redundant, or you'd rather me add the base path to existing tests instead. I'm happy to update it as you see fit.

File: packages/tests-unit/tests/core/routing/matcher.test.ts [hunk 1]
```diff
@@ -490,6 +490,21 @@ describe("fixDataPage", () => {
...
+    const event = createEvent({
+      url: "/base/_next/data/abc/test",
+    });
+
+    const response = fixDataPage(event, "abc");
+
+    expect(response.statusCode).not.toEqual(404);
+    expect(response).toEqual(event);
+
+    NextConfig.basePath = undefined;
```
Comment (sommeeeer) [inline packages/tests-unit/tests/core/routing/matcher.test.ts:505]: don't need this as it gets reset before each: https://github.com/opennextjs/opennextjs-aws/blob/a322aa699ac101a9bb46840c45edeefa6b661bb5/packages/tests-unit/tests/core/routing/matcher.test.ts#L39

Comment (sommeeeer) [inline packages/tests-unit/tests/core/routing/matcher.test.ts]: Don't need this one as `NextConfig` gets reset before each : https://github.com/opennextjs/opennextjs-aws/blob/a322aa699ac101a9bb46840c45edeefa6b661bb5/packages/tests-unit/tests/core/routing/matcher.test.ts#L39

**Review by conico974 — CHANGES_REQUESTED** [review]
https://github.com/opennextjs/opennextjs-aws/actions/runs/12284689715/job/34281243192?pr=672
Run `pnpm biome check --fix` at the root of the repo

**Comment by sommeeeer** [comment]
Looks like it failed on write permission reporting the test coverage in `actions/test`

**Comment by alebelcor** [comment]
So weird, the tests started failing locally for some reason now.

**Comment by conico974** [comment]
@alebelcor I think the reset are done on each describe, not on each test inside describe

**Comment by sommeeeer** [comment]
@alebelcor sorry about that, looks like conico is right. you could add ```ts
  beforeEach(() => {
    NextConfig.basePath = "";
  });``` under line 471, or just do what you did before my earlier review.

**Comment by sommeeeer** [comment]
```ts
beforeEach(()=> {
  NextConfig.basePath = undefined;
})```

if you add this under line 471 it seems to work here. might be that the global `beforeEach()` will only reset on each `describe`.

**Comment by sommeeeer** [comment]
if you add this under line 471 it seems to work here. might be that the global `beforeEach()` will only reset on each `describe`:

```ts
beforeEach(()=> {
  NextConfig.basePath = undefined;
})```

**Review by sommeeeer — APPROVED** [review]
LGTM, thanks!

**Review by conico974 — APPROVED** [review]
LGTM Thanks

**Pull request merged**